src/journal-remote/journal-upload.c

```diff
@@ -448,6 +490,57 @@ static void destroy_uploader(Uploader *u) {
         sd_event_unref(u->event);
 }
 
+#if LIBCURL_VERSION_NUM >= 0x075300
+static int update_content_encoding(Uploader *u, const char *accept_encoding) {
+        int r;
+
+        assert(u);
+
+        for (const char *p = accept_encoding;;) {
+                _cleanup_free_ char *encoding_value = NULL, *alg = NULL;
+                Compression algorithm;
+                CURLcode code;
+
+                r = extract_first_word(&p, &encoding_value, ",", 0);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to extract Accept-Encoding header value");
+                if (r == 0)
+                        break;
+
+                const char *q = encoding_value;
+                r = extract_first_word(&q, &alg, ";", 0);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to extract compression algorithm from Accept-Encoding header");
+
+                algorithm = compression_lowercase_from_string(alg);
+
+                FOREACH_ARRAY(opt, arg_compression.opts, arg_compression.size) {
+                        if (opt->algorithm != algorithm)
+                                continue;
+                        u->compression = *opt;
+
+                        _cleanup_free_ char *header = strjoin("Content-Encoding: ", compression_lowercase_to_string(u->compression.algorithm));
+                        if (!header)
+                                return log_oom();
+
+                        // Updating Content-Encoding header value
```

Comment:
Please use `/* ... */` style of comment.

Also, the current logic only updates the existing header. But, if the header is not found, we need to append new one.
```c
/* First, update existing Content-Encoding header. */
bool found = false;
for (...) {
        ...
}

/* If Content-Encoding header is not found, append new one. */
if (!found) {
        struct curl_slist *l = curl_slist_append(u->header, ...);
        if (!l)
                return log_oom();
        u->header = l;
}
```